### DISCUSSION TITLE
How to deal with the unconfirmed github issues?

### DISCUSSION THREAD

**Discussion by mcuee** [avrdudes/avrdude#1008] [General]
Right now the number of issues have some down to a more healthy number of around 100.

There are still 23 unconfirmed issues, how shall we deal with them?
https://github.com/avrdudes/avrdude/issues?q=is%3Aopen+is%3Aissue+label%3Aunconfirmed

Among them, quite some are related to xmega.

**Comment by mcuee** [comment]
Ref: xmega relared open issues (Updated on 24-September-2022)

Unconfirmed issues
No more.

Confirmed as bug
* https://github.com/avrdudes/avrdude/issues/365 
* https://github.com/avrdudes/avrdude/issues/1082 (on 21-Aug-2022)
* https://github.com/avrdudes/avrdude/issues/360 (on 24-Sept-2022)
* https://github.com/avrdudes/avrdude/issues/454 (on 24-Sept-2022)

Enhancement request:
* https://github.com/avrdudes/avrdude/issues/200 (enhancement)
* https://github.com/avrdudes/avrdude/issues/733 (enhancement request, with patch)

Recently closed issues:
* https://github.com/avrdudes/avrdude/issues/190 (should work with the right bootloader)
* https://github.com/avrdudes/avrdude/issues/164
* https://github.com/avrdudes/avrdude/issues/226
* https://github.com/avrdudes/avrdude/issues/243
* https://github.com/avrdudes/avrdude/issues/289
* https://github.com/avrdudes/avrdude/issues/304 
* https://github.com/avrdudes/avrdude/issues/310
* https://github.com/avrdudes/avrdude/issues/311
* https://github.com/avrdudes/avrdude/issues/314 
* *https://github.com/avrdudes/avrdude/issues/337
* https://github.com/avrdudes/avrdude/issues/391 (also affect other AVR chips, closed as invalid on 19-Aug-2022).
* https://github.com/avrdudes/avrdude/issues/404 
* https://github.com/avrdudes/avrdude/issues/435

**Comment by dl8dtl** [comment]
Thanks for all your (not only you but everyone involved) work dealing with all the old issues.

I've set #360 as an enhancement request, since it asks for a protocol addition.

I'm really not sure how relevant any AVR32 patches are, given that even Atmel/Microchip went away from them now (the EDBG chips were the last domain of UC3 devices). There's a patch, but I don't know of anyone who would be able to even verify it works.

I can try looking at the remaining unconfirmed items piecewise.

**Reply by mcuee** [reply]
Great that you can start to take a look at those unconfirmed items one by one.

And yes I think we can close the AVR32 related enhancement request based on your comments and the reality that Microchip will concentrate on ARM and PIC32 (even the MIPS based PIC32 is going lower priority) and stopped the development of AVR32 long ago.

**Reply by mcuee** [reply]
So I just closed the few AVR32 related feature requests.

* https://github.com/avrdudes/avrdude/issues/166
* https://github.com/avrdudes/avrdude/issues/239
* https://github.com/avrdudes/avrdude/issues/735
* https://github.com/avrdudes/avrdude/issues/977

**Comment by mcuee** [comment]
@dl8dtl 
Do we really need to take care of the long obsoleted AT90S parts?
* https://github.com/avrdudes/avrdude/issues/433 (AT90S8515)
* https://github.com/avrdudes/avrdude/issues/163 (AT90S2313)
* https://github.com/avrdudes/avrdude/issues/500 (AT90S8535)

**Reply by MCUdude** [reply]
I'd argue that we could drop support for them if we can get rid of legacy code written specifically for these parts. And if one _have_ to use one of these 20+ year-old parts, you could always just use an older version of Avrdude.

**Comment by mcuee** [comment]
Right now the un-confirmed issues have reduced to 13.
https://github.com/avrdudes/avrdude/issues?q=is%3Aopen+is%3Aissue+label%3Aunconfirmed

I think quite some of them may not be relevant any more. I took the liberty and closed them. Users can always re-open if there are similar issues happening again.
* #464 (I tested ft232r and no such issue)
* https://github.com/avrdudes/avrdude/issues/364 (I tested FT2232H and no such issue).
* #307 (I have not found such issue with STK500v2) --> it turns out the issue is real and replaced with #1073 and fixed by #1074.

**Comment by mcuee** [comment]
Linux gpio related issues:
* #455 (tested to be a real bug)
* #458 (duplicate of 455)
* #490 (I have tested it and no issue found -- so I closed it for now. Users can re-open if similar issues popped up).

**Comment by mcuee** [comment]
Right now the un-confirmed issues are related to xmega.
https://github.com/avrdudes/avrdude/issues?q=is%3Aopen+is%3Aissue+label%3Aunconfirmed

**Comment by mcuee** [comment]
Finally I am able to program the ATxmega32D4 breakout board using an AVRISP mkii clone. I am able to confirm that #365 is a bug and #243 is not valid (Closed).

**Comment by mcuee** [comment]
I closed #289 as invalid today. I just got an ATxmeag32A4U breakout board and there is no issue to flash the chip using an AVRISP mkii clone.

**Reply by mcuee** [reply]
I closed the very old issue #190 yesterday (27-Auguest-2022) since there are reports of successing using Arduino compatible bootloader for xmega.

**Comment by mcuee** [comment]
#391 was closed by @MCUdude two days ago.

**Comment by mcuee** [comment]
Adding new issue related to xmega with FLIP2 as a bug.
* https://github.com/avrdudes/avrdude/issues/1082

**Comment by mcuee** [comment]
As of today 28-August-2022, there are 76 issues remaining, 38 are enhancement requests, 26 are bugs, 5 are unconfirmed and 10 are related to documentation (some issues may have more than one label).

**Comment by mcuee** [comment]
@MCUdude 
Thanks for closing #310, #337 and #404 as invalid. Now we only have two unconfirmed issues, related to AVR109 bootloader for the ATxmega256A3 or similar xmega.
* #360 
* #454

Now that you have the ATxmega256A3BU Xplained board, please help to test the above two issues when you got a chance. Thanks,

**Reply by mcuee** [reply]
@MCUdude 
Thanks to confirm that #360 and #454 are indeed bugs. Now we do not have any unconfirmed issues. Great.
